tests/test_connect_roundtrip_and_edges.py

```diff
@@ -0,0 +1,237 @@
+"""round-trip and edge tests for ordinary vs extended CONNECT behavior."""
+
+from __future__ import annotations
+
+from typing import Iterable, List, Sequence, Tuple
+
+import pytest
+
+from h2.config import H2Configuration
+from h2.connection import H2Connection
+from h2.events import RequestReceived
+from h2.utilities import HeaderValidationFlags, validate_outbound_headers
+
+
+Headers = Sequence[Tuple[str | bytes, str | bytes]]
```

Comment:
Is this required here, or could it be re-used / imported from h2 itself? Otherwise I fear it might become outdated as both implementations diverge.